### PR TITLE
expose global info bar state

### PR DESCRIPTION
**Pull request by jamesgorrie**
Mixin of #5728

### CODE REVIEW

File: common/views/components/GlobalInfoBarContext/GlobalInfoBarContext.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  children: ReactNode;
+};
+type ContextProps = {
+  isVisible: boolean;
+  setIsVisible: (isVisible: boolean) => void;
+};
+const GlobalInfoBarContext = createContext<ContextProps>({
+  isVisible: false,
+  setIsVisible: () => {
+    // noop
```
Comment (ishouty) [inline common/views/components/GlobalInfoBarContext/GlobalInfoBarContext.tsx:13]: not setting visible ?

Comment (jamesgorrie) [reply common/views/components/GlobalInfoBarContext/GlobalInfoBarContext.tsx:13]: Do you mean the naming? The common pattern for the `useState` hook us `[value, setValue]`, [as you can see here](https://reactjs.org/docs/hooks-state.html). We tend to prefix `boolean`s with `is` e.g: `isCondition`.

Does this answer your question?

Comment (ishouty) [reply common/views/components/GlobalInfoBarContext/GlobalInfoBarContext.tsx:13]: ahh, its the default values which was confusing me. cause the context props wants a boolean but the setIsVisible doesnt expect anything.

File: common/views/components/InfoBanner/InfoBanner.tsx [hunk 2]
```diff
@@ -31,6 +34,12 @@ const InfoBanner: FunctionComponent<Props> = ({ cookieName, text }: Props) => {
     setIsVisible(false);
   };
 
+  useEffect(() => {
+    if (prevIsVisible !== isVisible && onVisibilityChange) {
+      onVisibilityChange(isVisible);
```
Comment (ishouty) [inline common/views/components/InfoBanner/InfoBanner.tsx:39]: so we have two states , one is more global which we update and can reference too.

Comment (jamesgorrie) [reply common/views/components/InfoBanner/InfoBanner.tsx:39]: We have the state of the `InfoBanner` component.

We give access to that property via a callback. The callback is passed from `GlobalInfoBannerContext` which then passes that onto anywhere that wants the value via `useContext`.

content model(1)

**Review by ishouty — APPROVED** [review]

**Pull request merged**